dev/NavigationView/NavigationView.cpp

```diff
@@ -1592,6 +1770,161 @@ void NavigationView::UpdateVisualStateForDisplayModeGroup(const winrt::Navigatio
     }
 }
 
+void NavigationView::OnNavigationViewItemTapped(const winrt::IInspectable& sender, const winrt::TappedRoutedEventArgs& args)
+{
+    if (auto nvi = sender.try_as<winrt::NavigationViewItem>())
+    {
+        if (IsSettingsItem(nvi))
+        {
+            OnSettingsInvoked();
+        }
+        else
+        {
+            OnNavigationViewItemInvoked(nvi);
+        }
+        nvi.Focus(winrt::FocusState::Pointer);
+    }
+}
+
+void NavigationView::OnNavigationViewItemKeyDown(const winrt::IInspectable& sender, const winrt::KeyRoutedEventArgs& args)
+{
+    auto key = args.Key();
+
+    if (auto nvi = sender.try_as<winrt::NavigationViewItem>())
+    {
+        if (IsSettingsItem(nvi))
+        {
+            // Because ListViewItem eats the events, we only get these keys on KeyDown.
+            if (key == winrt::VirtualKey::Space ||
+                key == winrt::VirtualKey::Enter)
+            {
+                args.Handled(true);
+                OnSettingsInvoked();
+            }
+        }
+        else
+        {
+            switch (args.Key())
```

Comment:
>switch [](start = 12, length = 6)

You aren't marking the args as handled in any of these cases, is that correct?

Reply:
Updated it to make it handled